components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/util/APIResourceMgtEndpointUtil.java

```diff
@@ -166,6 +168,21 @@ public static APIError handleAPIResourceMgtException(APIResourceMgtException e)
         return handleException(status, errorCode, e.getMessage(), e.getDescription());
     }
 
+    public static APIError handleAPIResourceCollectionMgtException(APIResourceCollectionMgtException e) {
+
+        Response.Status status = Response.Status.INTERNAL_SERVER_ERROR;
+        if (e instanceof APIResourceCollectionMgtClientException) {
+            LOG.debug(e.getMessage(), e);
+            status = Response.Status.BAD_REQUEST;
+        } else {
+            LOG.error(e.getMessage(), e);
+        }
+        String errorCode = e.getErrorCode(); //TODO: Update this
```

Comment:
Is this TODO needed

Reply:
No this error codes were defined in framework. TODO is removed.